contrib/setup-dashd.py

```
@@ -0,0 +1,168 @@
+            return
+        except Exception as e:
+            log(f"Download failed: {e}")
+            if attempt == retries:
+                sys.exit(f"Failed to download {url} after {retries} attempts")
+            time.sleep(5 * attempt)
+
+
+def extract(archive_path, dest_dir):
+    if archive_path.endswith(".zip"):
+        with zipfile.ZipFile(archive_path, "r") as zf:
+            zf.extractall(dest_dir)
```

Ruff (0.15.6)

[error] Uses of `tarfile.extractall()`

(S202)